Update file_sharing_configuration.adoc (#1675)
owncloud/docs: modules/admin_manual/pages/configuration/files/file_sharing_configuration.adoc

==== Restrict users to only share with users in their groups

> When this option is enabled, users can only share with other users that are members of groups that the user is a member of.

Reviewer: ```suggestion
When this option is enabled, users can only share with the users in the groups which they are a member of.
```
As suggested by @settermjd in the thread below.
IMO this reads better.

> They can still share with all groups of the instance but not with users that do not share a group.

Reviewer 2: Given the setting's wording, this option seems counter-intuitive, i.e., either they can **_only_** share with users in their groups, or they can do that, and share with other groups.

Reviewer 3: The difference here is sharing with **single** users or with **groups of users**. There are use cases where admins want to allow users to share with all single users in their instance but only with groups they are a member of. Or they should be restricted to share only with users in their groups but are allowed to share with all groups (e.g., without knowing exactly who is a member of such group). This is why this option has been split up some time ago.
If you want to restrict users to _really_ only be able to share within their groups, you need to tick both options.

Reviewer: @settermjd it is like @pmaier1 said. The challenge is how to describe it in English that is correct, understandable and brief - maybe all 3 cannot be achieved together. Suggestions are most welcome.

Reviewer 2: Thanks @pmaier1 @phil-davis. You're right, @phil-davis, wording is a little tricky. I'll see if I can come up with some meaningful suggestions.

Reviewer 2: Here's a mockup of a small rework of the two checkboxes.
The two checkboxes are now listed under an applicable sub-heading, inside a "Restrictions" group.

![new user and group restrictions wording](https://user-images.githubusercontent.com/196801/63439797-03eecf00-c42f-11e9-94e2-0149dd6e9749.png)

The intent here is to:

1. Reduce the number of required words
2. More intuitively show what the settings are about; and
3. Leave the door open to further restrictions in the future

Yes/no?

Reviewer 2: @phil-davis @jnweiger any feedback?

Author: "Users can only share with users in their group" is much better! Thank you!
But I still see some room for misinterpretation. 
All of short, precise, and still correct and easy english is hard.

How about mentioning group shares in the positive form, like in e.g. 
"Users can only share with users in their groups or with groups"

That makes the english a bit harder and longer, but (hopefully) adds precision.

I'd leave the final balancing to you. :-)

Reviewer 2: Thanks for the feedback, @jnweiger. I agree, conciseness, correctness, and precision in any language is hard (English is no friend in that respect). That said, I'll get it done.

Reviewer 2: Here's one final suggestion: "_Users can only share with the users in the groups which they are a member of_". I've been contemplating this for a while now, this morning, and I can't think of anything better. @jnweiger, would you be happy with it?

Reviewer: I am happy with @settermjd suggestion. I made a comment on the exact line where it would apply.

==== Restrict users to only share with users in their groups

> Check this option to confine sharing within group memberships.

Reviewer: Maybe this sentence could also be revised to be similar style to the one in the next section.
```suggestion
When this option is enabled, users can only share with other users that are members of groups that the user is a member of.
```
